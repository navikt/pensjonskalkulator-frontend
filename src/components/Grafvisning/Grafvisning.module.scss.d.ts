import globalClassNames from '../../style.d'
declare const classNames: typeof globalClassNames & {
  readonly buttons: 'buttons'
  readonly rightAligned: 'rightAligned'
  readonly tooltip: 'tooltip'
  readonly tooltipTable: 'tooltipTable'
  readonly tooltipTableHeaderCell: 'tooltipTableHeaderCell'
  readonly tooltipTableHeaderCell__left: 'tooltipTableHeaderCell__left'
  readonly tooltipTableHeaderCell__right: 'tooltipTableHeaderCell__right'
  readonly tooltipTableCell: 'tooltipTableCell'
  readonly tooltipTableCell__right: 'tooltipTableCell__right'
  readonly tooltipTableCellDot: 'tooltipTableCellDot'
  readonly tooltipLine: 'tooltipLine'
  readonly 'highcharts-tooltip-container': 'highcharts-tooltip-container'
  readonly 'highcharts-scrollable-mask': 'highcharts-scrollable-mask'
  readonly hidden: 'hidden'
}
export = classNames
