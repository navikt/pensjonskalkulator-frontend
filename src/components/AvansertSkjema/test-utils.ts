export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
