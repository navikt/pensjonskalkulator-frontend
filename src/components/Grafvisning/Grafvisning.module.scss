@import '../../scss/variables';

.buttons {
  display: flex;
  justify-content: space-between;
  height: var(--a-font-line-height-medium);
}

.rightAligned {
  flex-grow: 1;
  text-align: right;
}

.tooltip {
  right: 0.375rem;
  background-color: var(--a-white);
  border-bottom: 3px solid var(--a-gray-600);
  z-index: 3;

  & > span {
    position: relative !important;
    width: 100% !important;
    display: block;
  }

  &Table {
    position: relative;
    width: 100%;
    table-layout: fixed;
    background-color: var(--a-white);
    border-collapse: separate;
    border-spacing: 0 var(--a-spacing-1);
    z-index: 1;

    &HeaderCell {
      border-bottom: 1px solid var(--a-gray-600);
      font-size: var(var(--a-font-size-medium));

      &__left {
        width: 70%;
        white-space: break-spaces;
        font-weight: var(--a-font-weight-regular);
        text-align: left;
        padding-right: var(--a-spacing-3);

        @media (min-width: 370px) {
          width: auto;
          white-space: unset;
        }
      }

      &__right {
        font-weight: var(--a-font-weight-bold);
        text-align: right;
      }
    }

    &Cell {
      font-family: var(--a-font-family);
      font-size: var(--a-font-size-small);
      vertical-align: middle;

      &__right {
        font-weight: var(--a-font-weight-bold);
        text-align: right;
      }

      &Dot {
        height: var(--a-spacing-3);
        width: var(--a-spacing-3);
        border-radius: 50%;
        display: inline-block;
        vertical-align: middle;
        margin-right: var(--a-spacing-1);
        margin-top: -2px;
      }
    }
  }

  &Line {
    position: absolute;
    width: 2px;
    background-color: var(--a-gray-600);
  }
}

:global {
  .highcharts-tooltip-container {
    left: 1.125rem !important;
    right: 1.125rem;

    @media (min-width: $a-breakpoint-xl) {
      margin-left: auto;
      margin-right: auto;
      width: 100% !important;
      max-width: calc($a-breakpoint-xl-down - 2rem);
    }

    & > svg {
      display: none;
    }
  }

  .highcharts-scrollable-mask {
    filter: drop-shadow(-5px 0 3px rgb(255 255 255 / 25%));
  }
}

.hidden {
  display: none;
}