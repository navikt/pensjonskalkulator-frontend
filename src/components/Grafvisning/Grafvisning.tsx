import React, { useEffect, useState } from 'react'

import { ChevronLeftCircle, ChevronRightCircle } from '@navikt/ds-icons'
import { Button } from '@navikt/ds-react'
import Highcharts from 'highcharts'
import HighchartsReact from 'highcharts-react-official'

import {
  COLUMN_WIDTH,
  getAfpSeriesData,
  getAlderspensjonColumnOptions,
  getChartOptions,
  getInntektSeriesData,
  getPensjonsavtalerSeriesData,
  hasSeriesData,
  isSeriesColumnOptions,
  onVisFaerreAarClick,
  onVisFlereAarClick,
  removeHandleChartScrollEventListener,
} from './utils'

import styles from './Grafvisning.module.scss'
import clsx from 'clsx'

interface Props {
  aldere: string[]
  data: {
    alderspensjon: number[]
    afpPrivat: number[]
    inntekt: number[]
    pensjonsavtaler: number[]
  }
}

export const Grafvisning: React.FC<Props> = ({
  aldere,
  data: { alderspensjon, afpPrivat, inntekt, pensjonsavtaler },
}) => {
  const [showVisFlereAarButton, setShowVisFlereAarButton] = useState(false)
  const [showVisFaerreAarButton, setShowVisFaerreAarButton] = useState(false)

  const [chartOptions, setChartOptions] = useState<Highcharts.Options>(
    getChartOptions(styles, setShowVisFlereAarButton, setShowVisFaerreAarButton)
  )

  useEffect(() => {
    return removeHandleChartScrollEventListener
  }, [])

  useEffect(() => {
    setChartOptions({
      chart: {
        type: 'column',
        scrollablePlotArea: {
          minWidth: aldere.length * COLUMN_WIDTH * 1.6,
          scrollPositionX: 0,
        },
      },
      xAxis: {
        categories: aldere,
      },
      series: [
        hasSeriesData(inntekt) && getInntektSeriesData(inntekt),
        hasSeriesData(afpPrivat) && getAfpSeriesData(afpPrivat),
        hasSeriesData(alderspensjon) &&
          getAlderspensjonColumnOptions(alderspensjon),
        hasSeriesData(pensjonsavtaler) &&
          getPensjonsavtalerSeriesData(pensjonsavtaler),
      ].filter(isSeriesColumnOptions),
    })
  }, [alderspensjon, afpPrivat, inntekt, aldere])

  return (
    <section>
      <HighchartsReact highcharts={Highcharts} options={chartOptions} />
      <div className={styles.buttons}>
        <div className={clsx(!showVisFaerreAarButton && styles.hidden)}>
          <Button
            icon={<ChevronLeftCircle aria-hidden />}
            iconPosition="left"
            size="xsmall"
            variant="tertiary"
            onClick={onVisFaerreAarClick}
          >
            Vis færre år
          </Button>
        </div>
        <div
          className={clsx(
            styles.rightAligned,
            !showVisFlereAarButton && styles.hidden
          )}
        >
          <Button
            icon={<ChevronRightCircle aria-hidden />}
            iconPosition="right"
            size="xsmall"
            variant="tertiary"
            onClick={onVisFlereAarClick}
          >
            Vis flere år
          </Button>
        </div>
      </div>
    </section>
  )
}
