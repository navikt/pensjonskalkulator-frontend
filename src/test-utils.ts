import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
