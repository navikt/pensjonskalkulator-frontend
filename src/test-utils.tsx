import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
