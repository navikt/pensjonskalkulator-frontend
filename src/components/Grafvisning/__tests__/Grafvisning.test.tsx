import { render, screen } from '@/test-utils'
import { Grafvisning } from '@/components/Grafvisning/Grafvisning'

describe('Grafvisning', () => {
  it('rendrer som forventet', async () => {
    const aldere = ['62', '63', '64']
    const data = {
      alderspensjon: [100_000, 100_000, 100_000],
      afpPrivat: [10_000, 10_000, 10_000],
      inntekt: [200_000, 0, 0],
      pensjonsavtaler: [10_000],
    }

    const { container, asFragment } = render(
      <Grafvisning aldere={aldere} data={data} />
    )

    expect(await screen.findByText('Beregning')).toBeInTheDocument()

    expect(
      container.getElementsByClassName('highcharts-container')
    ).toHaveLength(1)
    expect(asFragment()).toMatchSnapshot()
  })

  it('rendrer bare legend for series med minst ett innslag som er hÃ¸yere enn 0', async () => {
    const aldere = ['62', '63', '64']
    const data = {
      alderspensjon: [100_000, 100_000, 100_000],
      afpPrivat: [0, 0, 0],
      inntekt: [200_000, 0, 0],
      pensjonsavtaler: [],
    }

    render(<Grafvisning aldere={aldere} data={data} />)

    await screen.findByText('Beregning')

    expect(screen.getByText('Inntekt', { exact: false })).toBeVisible()
    expect(screen.getByText('Alderspensjon', { exact: false })).toBeVisible()
    expect(
      screen.queryByText('Avtalefestet pensjon', { exact: false })
    ).not.toBeInTheDocument()
  })
})
