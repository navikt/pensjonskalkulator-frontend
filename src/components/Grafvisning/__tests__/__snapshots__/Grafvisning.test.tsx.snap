// Vitest Snapshot v1

exports[`Grafvisning > rendrer som forventet 1`] = `
<DocumentFragment>
  <section>
    <div
      data-highcharts-chart="0"
      style="overflow: visible;"
    >
      <div
        class="highcharts-scrolling-parent"
        style="position: relative;"
      >
        <div
          class="highcharts-fixed"
          dir="ltr"
          style="padding: 0px; margin: 0px; position: absolute; overflow: hidden; pointer-events: none; z-index: 2; top: 0px;"
        >
          <svg
            class="highcharts-root"
            height="400"
            style="font-family: Helvetica, Arial, sans-serif; font-size: 1rem;"
            version="1.1"
            viewBox="0 0 600 400"
            width="600"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc>
              Created with Highcharts 11.1.0
            </desc>
            <defs />
            <path
              class="highcharts-scrollable-mask"
              d="M 0 0"
              data-z-index="-1"
              fill="#ffffff"
              fill-opacity="0.85"
            />
            <g
              class="highcharts-legend highcharts-no-tooltip"
              data-z-index="7"
              style="pointer-events: auto;"
              transform="translate(0,-25)"
            >
              <rect
                class="highcharts-legend-box"
                fill="none"
                filter="none"
                height="19"
                rx="0"
                ry="0"
                stroke="#999999"
                stroke-width="0"
                width="84"
                x="0"
                y="0"
              />
              <g
                data-z-index="1"
              >
                <g>
                  <g
                    class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-0"
                    data-z-index="1"
                    transform="translate(0,-5)"
                  >
                    <rect
                      class="highcharts-point"
                      data-z-index="3"
                      fill="#868F9C"
                      height="14"
                      rx="7"
                      ry="7"
                      width="14"
                      x="1"
                      y="6"
                    />
                  </g>
                  <g
                    class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-1"
                    data-z-index="1"
                    transform="translate(21,-5)"
                  >
                    <rect
                      class="highcharts-point"
                      data-z-index="3"
                      fill="var(--a-purple-400)"
                      height="14"
                      rx="7"
                      ry="7"
                      width="14"
                      x="1"
                      y="6"
                    />
                  </g>
                  <g
                    class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-2"
                    data-z-index="1"
                    transform="translate(42,-5)"
                  >
                    <rect
                      class="highcharts-point"
                      data-z-index="3"
                      fill="var(--a-deepblue-500)"
                      height="14"
                      rx="7"
                      ry="7"
                      width="14"
                      x="1"
                      y="6"
                    />
                  </g>
                  <g
                    class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-3"
                    data-z-index="1"
                    transform="translate(63,-5)"
                  >
                    <rect
                      class="highcharts-point"
                      data-z-index="3"
                      fill="var(--a-green-400)"
                      height="14"
                      rx="7"
                      ry="7"
                      width="14"
                      x="1"
                      y="6"
                    />
                  </g>
                </g>
              </g>
            </g>
            <text
              class="highcharts-subtitle"
              data-z-index="4"
              style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666; pointer-events: auto;"
              text-anchor="middle"
              x="288"
              y="0"
            />
            <text
              class="highcharts-title"
              data-z-index="4"
              style="font-size: 20px; color: rgb(51, 51, 51); font-weight: bold; font-family: var(--a-font-family); fill: #333333; pointer-events: auto;"
              text-anchor="start"
              x="0"
              y="20"
            >
              Beregning
            </text>
          </svg>
          <div
            class="highcharts-legend highcharts-no-tooltip"
            style="position: absolute; left: 0px; top: -25px; opacity: 1; visibility: inherit; pointer-events: auto;"
          >
            <div
              style="position: absolute; left: 0px; top: 0px; opacity: 1;"
            >
              <div
                style="position: absolute; left: 0px; top: 0px; opacity: 1;"
              >
                <div
                  class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-0"
                  style="position: absolute; left: 0px; top: -5px; opacity: 1;"
                >
                  <span
                    data-z-index="2"
                    style="position: absolute; font-family: var(--a-font-family); font-size: 14px; white-space: nowrap; color: rgb(0, 0, 0); cursor: default; text-decoration: none; text-overflow: ellipsis; font-weight: regular; z-index: 0; overflow: hidden; margin-left: 0px; margin-top: 0px; left: 21px; top: 5px; z-index: 0;"
                  >
                    Inntekt (l√∏nn m.m.)
                  </span>
                </div>
                <div
                  class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-1"
                  style="position: absolute; left: 21px; top: -5px; opacity: 1;"
                >
                  <span
                    data-z-index="2"
                    style="position: absolute; font-family: var(--a-font-family); font-size: 14px; white-space: nowrap; color: rgb(0, 0, 0); cursor: default; text-decoration: none; text-overflow: ellipsis; font-weight: regular; z-index: 0; overflow: hidden; margin-left: 0px; margin-top: 0px; left: 21px; top: 5px; z-index: 0;"
                  >
                    Avtalefestet pensjon (AFP)
                  </span>
                </div>
                <div
                  class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-2"
                  style="position: absolute; left: 42px; top: -5px; opacity: 1;"
                >
                  <span
                    data-z-index="2"
                    style="position: absolute; font-family: var(--a-font-family); font-size: 14px; white-space: nowrap; color: rgb(0, 0, 0); cursor: default; text-decoration: none; text-overflow: ellipsis; font-weight: regular; z-index: 0; overflow: hidden; margin-left: 0px; margin-top: 0px; left: 21px; top: 5px; z-index: 0;"
                  >
                    Alderspensjon (NAV)
                  </span>
                </div>
                <div
                  class="highcharts-legend-item highcharts-column-series highcharts-color-undefined highcharts-series-3"
                  style="position: absolute; left: 63px; top: -5px; opacity: 1;"
                >
                  <span
                    data-z-index="2"
                    style="position: absolute; font-family: var(--a-font-family); font-size: 14px; white-space: nowrap; color: rgb(0, 0, 0); cursor: default; text-decoration: none; text-overflow: ellipsis; font-weight: regular; z-index: 0; overflow: hidden; margin-left: 0px; margin-top: 0px; left: 21px; top: 5px; z-index: 0;"
                  >
                    Pensjonsavtaler (arbeidsgiver)
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="highcharts-scrolling"
          style="overflow-x: auto; overflow-y: hidden; height: 400px;"
        >
          <div
            class="highcharts-inner-container"
          >
            <div
              class="highcharts-container "
              dir="ltr"
              id="highcharts-0"
              style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none; outline: none;"
            >
              <svg
                aria-label="Beregning"
                class="highcharts-root"
                height="400"
                role="img"
                style="font-family: Helvetica, Arial, sans-serif; font-size: 1rem;"
                version="1.1"
                viewBox="0 0 600 400"
                width="600"
                xmlns="http://www.w3.org/2000/svg"
              >
                <desc>
                  Created with Highcharts 11.1.0
                </desc>
                <defs>
                  <filter
                    id="highcharts-drop-shadow-0"
                  >
                    <fedropshadow
                      dx="1"
                      dy="1"
                      flood-color="#000000"
                      flood-opacity="0.75"
                      stdDeviation="2.5"
                    />
                  </filter>
                  <clippath
                    id="highcharts-1-"
                  >
                    <rect
                      fill="none"
                      height="NaN"
                      width="NaN"
                      x="0"
                      y="0"
                    />
                  </clippath>
                  <clippath
                    id="highcharts-5-"
                  >
                    <rect
                      fill="none"
                      height="NaNNaNNaNNaN"
                      width="NaNNaNNaNNaN"
                      x="0"
                      y="0"
                    />
                  </clippath>
                </defs>
                <rect
                  class="highcharts-background"
                  fill="#ffffff"
                  filter="none"
                  height="400"
                  rx="0"
                  ry="0"
                  width="600"
                  x="0"
                  y="0"
                />
                <rect
                  class="highcharts-plot-background"
                  fill="none"
                  filter="none"
                  height="NaN"
                  width="NaN"
                  x="NaN"
                  y="0"
                />
                <rect
                  class="highcharts-plot-border"
                  data-z-index="1"
                  fill="none"
                  height="385"
                  stroke="#cccccc"
                  stroke-width="0"
                  width="725"
                  x="0"
                  y="0"
                />
                <g
                  class="highcharts-grid highcharts-xaxis-grid"
                  data-z-index="1"
                >
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="none"
                    stroke-width="0"
                  />
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="none"
                    stroke-width="0"
                  />
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="none"
                    stroke-width="0"
                  />
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="none"
                    stroke-width="0"
                  />
                </g>
                <g
                  class="highcharts-grid highcharts-yaxis-grid"
                  data-z-index="1"
                >
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="4,3"
                    stroke-width="1"
                  />
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="4,3"
                    stroke-width="1"
                  />
                  <path
                    class="highcharts-grid-line"
                    d="M 0 0"
                    data-z-index="1"
                    fill="none"
                    opacity="1"
                    stroke="#e6e6e6"
                    stroke-dasharray="4,3"
                    stroke-width="1"
                  />
                </g>
                <g
                  class="highcharts-axis highcharts-xaxis"
                  data-z-index="2"
                >
                  <path
                    class="highcharts-axis-line"
                    d="M 0 0"
                    data-z-index="7"
                    fill="none"
                    stroke="#333333"
                    stroke-width="1"
                  />
                </g>
                <g
                  class="highcharts-axis highcharts-yaxis"
                  data-z-index="2"
                >
                  <text
                    class="highcharts-axis-title"
                    data-z-index="7"
                    style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
                    text-anchor="end"
                    transform="translate(0,0)"
                    x="NaN"
                    y="24"
                  >
                    Tusen kroner
                  </text>
                  <path
                    class="highcharts-axis-line"
                    d="M 0 0"
                    data-z-index="7"
                    fill="none"
                    stroke="#333333"
                    stroke-width="0"
                  />
                </g>
                <g
                  class="highcharts-series-group"
                  data-z-index="3"
                  filter="none"
                >
                  <g
                    class="highcharts-series highcharts-series-0 highcharts-column-series highcharts-tracker"
                    clip-path="url(#highcharts-5-)"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 0.001)"
                  />
                  <g
                    class="highcharts-markers highcharts-series-0 highcharts-column-series"
                    clip-path="none"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 1)"
                  />
                  <g
                    class="highcharts-series highcharts-series-1 highcharts-column-series highcharts-tracker"
                    clip-path="url(#highcharts-5-)"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 0.001)"
                  />
                  <g
                    class="highcharts-markers highcharts-series-1 highcharts-column-series"
                    clip-path="none"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 1)"
                  />
                  <g
                    class="highcharts-series highcharts-series-2 highcharts-column-series highcharts-tracker"
                    clip-path="url(#highcharts-5-)"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 0.001)"
                  />
                  <g
                    class="highcharts-markers highcharts-series-2 highcharts-column-series"
                    clip-path="none"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 1)"
                  />
                  <g
                    class="highcharts-series highcharts-series-3 highcharts-column-series highcharts-tracker"
                    clip-path="url(#highcharts-5-)"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 0.001)"
                  />
                  <g
                    class="highcharts-markers highcharts-series-3 highcharts-column-series"
                    clip-path="none"
                    data-z-index="0.1"
                    opacity="1"
                    transform="translate(NaN,44) scale(1 1)"
                  />
                </g>
                <text
                  class="highcharts-caption"
                  data-z-index="4"
                  style="color: rgb(102, 102, 102); font-size: 0.8em; fill: #666666;"
                  text-anchor="start"
                  x="0"
                  y="387"
                />
                <g
                  class="highcharts-axis-labels highcharts-xaxis-labels"
                  data-z-index="7"
                >
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="middle"
                    transform="translate(0,0)"
                    x="0"
                  >
                    62
                  </text>
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="middle"
                    transform="translate(0,0)"
                    x="0"
                  >
                    63
                  </text>
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="middle"
                    transform="translate(0,0)"
                    x="0"
                  >
                    64
                  </text>
                </g>
                <g
                  class="highcharts-axis-labels highcharts-yaxis-labels"
                  data-z-index="7"
                >
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="end"
                    transform="translate(0,0)"
                    x="0"
                  >
                    0
                  </text>
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="end"
                    transform="translate(0,0)"
                    x="0"
                  >
                    200
                  </text>
                  <text
                    style="color: rgb(51, 51, 51); cursor: default; font-size: 0.8em; fill: #333333;"
                    text-anchor="end"
                    transform="translate(0,0)"
                    x="0"
                  >
                    400
                  </text>
                </g>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_buttons_e19f10"
    >
      <div
        class="_hidden_e19f10"
      >
        <button
          class="navds-button navds-button--tertiary navds-button--xsmall"
        >
          <span
            class="navds-button__icon"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12 21.25a9.25 9.25 0 1 0 0-18.5 9.25 9.25 0 0 0 0 18.5ZM22.75 12c0 5.937-4.813 10.75-10.75 10.75S1.25 17.937 1.25 12 6.063 1.25 12 1.25 22.75 6.063 22.75 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="m8.44 12 4.53-4.53 1.06 1.06L10.56 12l3.47 3.47-1.06 1.06L8.44 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            aria-live="polite"
            class="navds-label navds-label--small"
          >
            Vis f√¶rre √•r
          </span>
        </button>
      </div>
      <div
        class="_rightAligned_e19f10 _hidden_e19f10"
      >
        <button
          class="navds-button navds-button--tertiary navds-button--xsmall"
        >
          <span
            aria-live="polite"
            class="navds-label navds-label--small"
          >
            Vis flere √•r
          </span>
          <span
            class="navds-button__icon"
          >
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12 2.75a9.25 9.25 0 1 0 0 18.5 9.25 9.25 0 0 0 0-18.5ZM1.25 12C1.25 6.063 6.063 1.25 12 1.25S22.75 6.063 22.75 12 17.937 22.75 12 22.75 1.25 17.937 1.25 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="m15.56 12-4.53 4.53-1.06-1.06L13.44 12 9.97 8.53l1.06-1.06L15.56 12Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
